lib/analysis.js

```diff
@@ -80,6 +82,16 @@ AnalysisFactory.prototype.create = function(configuration, definition, callback)
                     return done(err, analysis);
                 });
             },
+            function analysis$collectRequirements(analysis, done) {
+                requirements.computeRequirements(analysis, function(err) {
+                    return done(err, analysis);
+                });
+            },
+            function analysis$validateRequirements(analysis, done) {
```

Comment:
I'm changing all this so we get the requirements and validate in a single operations, and do it before registering, as we have talked about.